pyparsing/helpers.py

```diff
@@ -55,7 +55,7 @@ def countedArray(expr, intExpr=None):
 
     def countFieldParseAction(s, l, t):
         n = t[0]
-        arrayExpr << And([expr] * n) if n else empty
+        arrayExpr << (And([expr] * n) if n else empty)
```

Comment:
Can be written as just `arrayExpr << (expr * n) if n else empty` (the parens might not be needed, but I think they help readability).  I generally dislike using the explicit And constructions if they can be avoided., I much prefer using '+', '-', and '*' operators.

Otherwise looks good, good catch in fixing the unit tests! (another flag that we are changing the API...)

Reply:
Actually, the expr * n was also my first choice, but I though that it was like that for some reason. Actually the `__mul__` in `ParserElement` has some overhead, so probably explicit `And` in this case may be faster (especially here, as this is called for each parsed "instance" of such array in parsed data), so I've left this unchanged.
Also the parentheses around are needed as left shift operator has lower precedence than ternary if/else.